File: pet-boarding/pet-boarding-frontend/src/api/attendance.js
import request from '@/utils/request'

export function getAttendanceList(params) {
  return request.get('/attendance/list', { params })
}

export function clockIn(data) {
  return request.post('/attendance/clockin', data)
}

export function clockOut(data) {
  return request.post('/attendance/clockout', data)
}

File: pet-boarding/pet-boarding-backend/src/main/java/com/petboarding/entity/Attendance.java
package com.petboarding.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("attendance")
public class Attendance {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long employeeId;

    private Long shopId;

    private LocalDate date;

    private LocalDateTime clockIn;

    private LocalDateTime clockOut;

    private Integer status;

    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}
